app/src/main/kotlin/com/wire/android/media/audiomessage/ConversationAudioMessagePlayer.kt

```diff
@@ -34,14 +36,17 @@ import kotlinx.coroutines.flow.filterNotNull
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.merge
+import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.launch
 import javax.inject.Inject
+import javax.inject.Singleton
 
+@Singleton
```

Comment:
making it a Singleton it not really the way to go since we do not want to keep it in memory forever maybe try @Reusable instead ?

Reply:
I think there is potential for many resusable being used at the same time, when threads gets involved. Not sure if that matters in this case.

Reply:
unfortunately `@Reusable` is useless here, as it doesn't guaranty "not re-creating" a new instance of Player. So i added a Singleton Provider, which will provide a player instance (create it if needed) and free space, when player is not used anymore